src/cargo/util/command_prelude.rs

```diff
@@ -694,13 +702,17 @@ impl<'a> ArgMatchesExt for ArgMatches<'a> {
     fn _is_present(&self, name: &str) -> bool {
         self.is_present(name)
     }
+
+    fn _is_valid_arg(&self, name: &str) -> bool {
+        self.is_valid_arg(name)
+    }
```

Comment:
> ArgMatches now panics when accessing an undefined argument but clap
takes advantage of that for sharing code across commands that have
different subsets of arguments defined. I've extended clap so we can
"look before you leap" and put the checks at the argument calls to
start off with so its very clear what is tenuously shared. This
allows us to go in either direction in the future, either addressing
how we are sharing between commands or by moving this down into the
extension methods and pretending this clap feature doesn't exist